internal/tracker/client.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package tracker
+
+import (
+	"context"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/tektoncd/pipeline/pkg/remote/oci"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type Client interface {
+	GetTektonObject(ctx context.Context, bundle, kind, name string) (runtime.Object, error)
```

Comment:
The interface does return a runtime.Object. I can rename it to `GetTektonPipelineObject` but that seems misleading. To do, I think it would be best to change the interface to return a Pipeline instead of an Object. However, that would require adding a bit of code in `exoClient.GetTektonObject` to perform the conversion. I think we should avoid doing that because that code is not covered by unit tests only by acceptance tests.

I'm inclined to leave it as is unless you feel strongly about changing.

Reply:
My main concern is that this is a leaky abstraction and a bit overly generic. Let's leave it as is and refactor if we feel like it.